sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java

```diff
@@ -35,5 +40,18 @@
      */
     DefaultAzureCredential(ArrayDeque<TokenCredential> tokenCredentials) {
         super(tokenCredentials);
+        this.tokenCredentials = new ArrayList<TokenCredential>(tokenCredentials.size());
+        this.tokenCredentials.addAll(tokenCredentials);
+    }
+
+    /**
+     * Get the list of credentials sequentially used by {@link DefaultAzureCredential} to attempt authentication.
+     * The credentials in the returned list and their order may change in future versions of Identity.
+     * This API is not intended to be used in production ready code and should only be used for development purposes.
+     *
+     * @return The list of {@link TokenCredential}.
+     */
+    public List<TokenCredential> getCredentials() {
+        return Collections.unmodifiableList(tokenCredentials);
```

Comment:
Why returning an unmodifiable list here? I thought the intention of this API is return a copy of the current list for the user to modify before creating a CTC out of it.

Reply:
I think @schaabs said, we either return immutable list or a copy of it.
I think having it immutable prevents the use of indexes to replace/re-order items in place. and will require users to create their custom list whenever they want to modify it.
We can return a copy too.

Reply:
returns a copy now.